static/app/views/issueList/savedIssueSearches.tsx

```diff
@@ -47,18 +48,24 @@ const SavedSearchItem = ({
   const {mutate: deleteSavedSearch} = useDeleteSavedSearchOptimistic();
   const hasOrgWriteAccess = organization.access?.includes('org:write');
 
+  const canEdit =
+    savedSearch.visibility === SavedSearchVisibility.Owner || hasOrgWriteAccess;
+
   const actions: MenuItemProps[] = [
     {
       key: 'edit',
       label: 'Edit',
-      disabled: true,
-      details: 'Not yet supported',
+      disabled: !canEdit,
+      details: !canEdit ? t('You do not have permission to edit this search.') : '',
+      onAction: () => {
+        openModal(deps => (
+          <EditSavedSearchModal {...deps} {...{organization, savedSearch}} />
+        ));
+      },
     },
     {
-      disabled: !hasOrgWriteAccess,
-      details: !hasOrgWriteAccess
-        ? t('You do not have permission to delete this search.')
-        : '',
+      disabled: !canEdit,
+      details: !canEdit ? t('You do not have permission to delete this search.') : '',
```

Comment:
```suggestion
      details: !canEdit ? t('You do not have permission to delete this search.') : undefined,
```

Maybe more appropriate?

Reply:
Same above